src/server/game/Handlers/MailHandler.cpp

```diff
@@ -22,8 +22,11 @@ bool WorldSession::CanOpenMailBox(uint64 guid)
 {
     if (guid == _player->GetGUID())
     {
-        sLog->outError("%s attempt open mailbox in cheating way.", _player->GetName().c_str());
-        return false;
+        return true;
```

Comment:
why changing this return?

Reply:
to allow open mailbox in cheating way i guess

Reply:
it must be something like:

```suggestion
if (guid == _player->GetGUID())
{
  QueryResult result_prem = CharacterDatabase.PQuery("SELECT * FROM premium WHERE active = 1 AND AccountId = %u", _player->GetSession()->GetAccountId());
  if (!result_prem) {
    sLog->outError("%s attempt open mailbox in cheating way.", _player->GetName().c_str());
    return false;
  } else { 
    return true;
}
```

Reply:
What is the premium-module you are talking about?